### ISSUE TITLE
Low/High flap thresholds should have a default of null

### ISSUE THREAD

**Issue by amdprophet** [sensu/sensu-go#1235]
Low & high flap thresholds currently default to 0. We cannot determine if the value is unset or was intentionally set to 0.

## Expected Behavior
Low / high flap thresholds should allow for (and default to) null values.

## Current Behavior
Low / high flap thresholds default to 0.

## Possible Solution
Change to a type that allows for null values (int pointer?).

## Steps to Reproduce (for bugs)
N/A

## Context
N/A

## Your Environment
N/A

**Comment by grepory** [comment]
Implement default in the API.

**Comment by grepory** [comment]
If we default to -1, then that's an invalid value, but let's just change validation to include accepting this invalid value to indicate that it wasn't set.

**Comment by echlebek** [comment]
Looks like the protos used uint values, so we'll have to change the proto regardless.

I'm just going to add nullable=true, so that we don't have to change the actual wire format.

**Comment by echlebek** [comment]
Last night as I was drifting off to sleep, I worked out a way of doing this without changing anything! Predictably I forgot about it again until just now.

So: we can consider `low_flap_threshold == 0 && high_flap_threshold == 0` to be a "disabled" state for flap detection.

Consider that these values are two ends of a range. If we consider `low_flap_threshold` to be inclusive, and `high_flap_threshold` to be exclusive, then `(0, 0)` is an invalid range and can be considered a null state.

All other possible combinations such that `low_flap_threshold < high_flap_threshold` are valid.

I'm going to amend the PR I made to have this validation, and mention flap detection in the docs.
